### ISSUE TITLE
Some bugs     [sample code from ./docs]

### ISSUE THREAD

**Issue by yizhangliu** [jina-ai/docarray#466]
```python
from docarray import DocumentArray,Document

da = DocumentArray().empty(3)
for d in da:
       d.chunks = DocumentArray.empty(2)
       d.matches = DocumentArray.empty(2)
    
print(len(da['@m']), len(da['@m[:3]']), len(da['@m[1:3]']))        #results：  6  6  3.   '@m[:3]'=6 ?
print(da[..., 'id'])             # IndexError: Unsupported index type builtins.tuple: (Ellipsis, 'id')
print(da[...][:, 'id'])          # It's OK.    length=15
```

**Comment by JoanFM** [comment]
Hey @yizhangliu ,

Thanks for reporting the issue.

Let me explain a couple of things.

- For the problem with `da[..., 'id']` we are working on a fix in #467 
- Then why `len(da[@m[:3]])` gives 6. Is becasuse this [:3] slice is applied at every match, and not at the complete list of matches. Therefore doing @m[:3] lets the user find 2 matches per doc leading to 6 as a result. While doing @m[1:3] finds only a single match per root document, and therefore leading to 3 matches. 

You may understand better the difference by seeing this:


```python
from docarray import DocumentArray,Document

da = DocumentArray().empty(4)
for d in da:
       d.chunks = DocumentArray.empty(2)
       d.matches = DocumentArray.empty(2)

print(len(da['@m']), len(da['@m[:3]']), len(da['@m[1:3]']))        #results：  8 8 4
```

**Comment by yizhangliu** [comment]
@JoanFM  Thanks.
